mountpoint-s3/tests/fs.rs

```diff
@@ -1482,6 +1482,45 @@ async fn test_readdir_rewind_with_local_files_only() {
     assert_eq!(new_entries.len(), 3); // 1 new local file + 2 dirs (. and ..) = 3 entries
 }
 
+#[tokio::test]
+async fn test_readdir_repeat_response_partial() {
+    let (client, fs) = make_test_filesystem("test_readdir_repeat_response", &Default::default(), Default::default());
+
+    for i in 0..48 {
+        // "." and ".." make it a round 50 in total
+        client.add_object(&format!("foo{i}"), b"foo".into());
+    }
+
+    let dir_handle = fs.opendir(FUSE_ROOT_INODE, 0).await.unwrap().fh;
+    let max_entries = 25;
+
+    // FUSE( 10) READDIRPLUS fh FileHandle(1), offset 0, size 4096
+    // first request should just succeed
+    let first_response = ls(&fs, dir_handle, 0, max_entries).await;
+    assert!(first_response.len() == 25);
+
+    // FUSE( 11) INTERRUPT unique RequestId(10)
+    // don't really interrupt the application, but emulate the resulting fuse requests
+
+    // FUSE( 12) READDIRPLUS fh FileHandle(1), offset 1, size 4096
+    // first response was partially discarded because of an interrupt, but the returned entries should be the same for the second request
+    let second_response = ls(&fs, dir_handle, 1, max_entries).await;
+    assert_eq!(&first_response[1..], &second_response[..]);
```

Comment:
I'd prefer we generalize over this (`test_case(..)`) and provide some edge cases.

In particular, I'm looking for:
- `1` as you have
- `24`, i.e right at the end of the page so we know we're not doing anything wrong on the edges